### DISCUSSION TITLE
Modelling bus-like system like Powerline, etc. in Netbox

### DISCUSSION THREAD

**Discussion by jonasjelonek** [netbox-community/netbox#14252] [Q&A]
Hey everyone!

I'm completely new to Netbox but just installed it in my network and want to get familiar with it by modelling my home network with it. This was quite straightforward so far, reading a bit through the documentation and adding my router, switches, connections, etc. worked really well so far.
However, I came across one thing that is IMO hard to model in Netbox: PLC / Powerline connection. I have to use this in my home atm for some connections.  PLC can be thought of like a bus system whereas the regular Ethernet topology is usually rather point-to-point, especially when modelling in Netbox. But I wouldn't restrict this to PLC only, I also could think of other bus systems to be modelled in Netbox, e.g. Modbus, KNX, etc.

In my home I have four PLC adapters which are all connected to each other via the houses power cabling. In Netbox, I modelled each of these PLC adapters with two interfaces, one for the Ethernet with which it is connected to my router/a device, and the other one for the "PLC network". Both interfaces are bridge to each other. Everything fine up to this point. 
For the PLC interface I would need something like a bus entity or be able to connected the PLC interface of one adapter to those of all the other adapters. However that doesn't seem to be possible because only one interface can be connected to another at a time.

I tried two things so far:
1. Using Rear and Front ports to "split the cable". I read about this in the documentation however I am not sure if I understood this completely.
Here I have on every PLC adapter one Rear Port, which is connected to the PLC interface, and then 3 Front ports which are "mapped" to the Rear port. Then I connect all front ports of all PLC adapters with each other. This works so far for this small setup, however it doesn't work so well with the cable tracing. When I click on "Trace" on the any of the front ports, I don't get a path. Doing this on the Rear port just says that the path is split, and by clicking on one of the splitted paths I just get redirected to the tracing for the Front ports again.
Besides these problems, I think this can get really messy if there are more PLC adapters, and it feels like this isn't the correct modelling for this.

2. Adding a dummy device for the PLC-Bus with several interfaces which connect to each PLC adapters. This could then be thought of a hub/bridge to which all PLC adapters are connected and are therefore (at least in theory) connected to each other.
At least I have a better feeling of this as a modelling approach, although not quite optimal. And the cable tracing seems to work a bit better, at least some paths are better recognized.

I am also not sure if I got correctly what the cable tracing is supposed to be/do. It shows only partial paths for me up to a specific point. And I would think of such a functionality like 'select two devices in your network and show the path between both', however this doesn't seem to be possible or even intended?

Thanks for reading up to this point, I hope someone can help me.
Greetings

**Comment by dawnbenton** [comment]
You just described exactly my evening so far and this is exactly where I just stopped five minutes ago to ponder this very thing. My search led to this post.

Did you decide to go forward with the dummy hub device? I briefly thought about using "circuits", "front/back ports", and "L2VPNs" as that seemed like a close way to do it -- but a dummy/hub device seems to make the most sense and is the approach I was about to take -- I'm curious to know if you ran into any gotchas? 

Thank you.

**Reply by jonasjelonek** [reply]
Hi, thanks for your comment!

So I decided to keep that way with the dummy hub device. I tried this front/back ports stuff before, and it was ok but I had problems then with the cable tracing, and this doesn't scale very well because you have to connect each device to all the others, leading to a lot of connections if your PLC network is relatively big.

Then I had the dummy hub idea, which is quite obvious since PLC is somehow a hub. Setting this up went pretty well, was quite easy and low effort, and looks fine and accurate to me. Also the cable tracing view improved with that, although still not good imho.
I'm also using the [topology views plugin](https://github.com/netbox-community/netbox-topology-views) to get a proper network overview. And there it's at least imo accurately shown ... as a hub device to which all PLC adapters are connected.

**Comment by nklamann** [comment]
I have the same problem and wonder about the hub idea . 

Does such a virtual hub have rear ports or front ports ?

And how do you formulate the fact that they **all** are connected with each other ?

**Reply by jonasjelonek** [reply]
I didn't use front and rear ports now, not sure if this is possible but it was too complicated to dive into that. Right now, I just have a dummy device with several interfaces, one interface for each Powerline adapter in my network. And within the dummy device, they are "connected" by a bridge interface.

Sadly, connections don't traverse bridges so I haven't really found a way for Netbox to recognize that they are really all connected. But it's the same issue with normal networking when using bridges.

So the interconnection between them is in the end somewhat a blackbox device to which the Powerline adapters are just connected, and it is enough for me to model this.

**Comment by nklamann** [comment]
I use PowerLAN as a Point-to-Point connection, i.e. i use the Adapters in pairs. 

Their device type has 2 (in my case) Front ports of type copper and one Rear Port of type power. Both Front ports are connected to this rear port. This expresses the fact, that it does not matter which of the Front ports is used. 

Then I model 2 connectors of this type and connect their rear ports. Done.

Trace works then.
